src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,30 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, for example: /quotas/users?minOccupationRatio=0.0&maxOccupationRatio=0.0, the search feature is supposed to return newly created users
+who have not received any email yet at that point. However, this is not the case because the quotas are currently being initialized only after
+a user receives the first email.
+
+We need to initialize user quotas upon user creation time. The problem is: there is currently no event at user creation 
+and since the quota-search feature is a plugin of James, it cannot be hardwired into the domain logic of user management to initialize the quota for a just created user.
+
+## Decision
+
+For quota-search to be initialized/removed for a given user while keeping this feature as a plugin, we decided to adopt the Event Driven pattern we already use in mailbox-api. 
+We can create new events related to user management (UserCreated, UserRemoved and so on).
+
+To achieve that, we will extract the EventBus out of mailbox-api in order to make it a utility component (eventbus-api), then we make both mailbox-api and data-api depend on that new module. 
+
+## Consequences
+
+Mailbox-api would leverage the EventBus to keep exposing the MailboxListener api without changes on top of the generified EventBus. 
+We can no longer use the sealed hierarchy for events as they depend on different specific components. We need to define an common Event in eventbus-api, then create new implementation related to the domain where we want to make use of the EventBus. 
+
+We could rely on the EventBus reliability for building any feature in James. Breaking changes would likely happen on the MailboxEvent serialization. Upgrade will need to happen with an empty EventBus & DeadLetter.
```

Comment:
With @mbaechler proposal you adopted it is no longer true: ` Breaking changes would likely happen on the MailboxEvent serialization. Upgrade will need to happen with an empty EventBus & DeadLetter.`

Let's remove it.